_posts/2018-09-13-API-Gateway-Types-Compared.md

```diff
@@ -0,0 +1,153 @@
+
+|  Ireland      | Edge | Regional | Private |
+|--------------:|:----:|:--------:|:-------:|
+|  Min          |  31  |   90     |  96     |
+|  Mean         |  62  |   105    |  130    |
+|  **Median**   |**61**| **103**  |**104**  |
+|  Max          |  226 |   214    |  25036  |
+
+|  N. Virginia  | Edge  | Regional | Private |
+|--------------:|:-----:|:--------:|:-------:|
+|  Min          |  100  |   352    |  356    |
+|  Mean         |  110  |   356    |  364    |
+|  **Median**   |**104**| **355**  |**360**  |
+|  Max          |  1114 |   560    |  23498  |
+
+|  São Paulo    | Edge  | Regional | Private |
+|--------------:|:-----:|:--------:|:-------:|
+|  Min          |  236  |   901    |  903    |
+|  Mean         |  252  |   910    |  916    |
+|  **Median**   |**241**| **909**  |**915**  |
+|  Max          |  766  |   1120   |  1918   |
+
+## Conclusions
+A few things immediately jump out:
+1. Edge Optimized API Gateways are **fast**.
+1. A Private API Gateway is generally as fast as a Regional API Gateway.
+1. Private API Gateways over VPC peering are less stable than the other variants.
+1. Connecting to a Regional API Gateway from the same region generated the lowest *minimal* response time, but in our test, Edge Optimized was still faster on average.
+
+All in all, we find the results quite surprising! 
+
+**Point 1**: We had not expected the Edge Optimized API Gateway to be so much more performant than the other two types. Based on this we would advise to use Edge Optimized over Regional in most use cases. If you are using Regional, do place your own CloudFront distribution in front of it.
+
+We can't say for sure *why* Edge Optimized is so much faster. In the end, traffic from the Edge locations will still need to travel to the API Gateway. It's unlikely that these paths are faster than other internal Amazon traffic, such as VPC Peerings. Other possible reasons for the fast Edge Optimized API are:
+* There is still some caching taking place, even though caching has been disabled.
+* Edge Optimized API Gateway has some internal smarts that allow certain elements to be executed on the Edge locations. We've executed our tests using Mock responses, and these could very well be delegated.
+
+But if either of these mechanisms are in place, we would expect the Edge Optimized tests to be equally fast from all different regions. And we're not seeing that either. In fact, Frankfurt is fastest, followed by Ireland, followed by North Virginia, followed by São Paulo, which confirms that distance plays a role.
+
+**Point 2**: Regional and Private gateways are in almost all situations equally fast. We did not expect this, because the traffic for the Private API Gateway stays within the Amazon networks, which we expected to be a bit faster. Of course, because we were using EC2 instances in our tests, it's well possible that the traffic for the Regional API Gateway *also* stayed within the Amazon network, which might explain this finding.
```

Comment:
`Of course, because we were using EC2 instances in our tests, it's well possible that the traffic for the Regional API Gateway *also* stayed within the Amazon network, which might explain this finding.` Duhhh. 🤓 AWS is not going to route its own ASN blocks over the internet and back. Their routers are smart enough to keep it in-house.

Reply:
Sure, but still it's going to be a different route with more hops.

Reply:
Ok